docs/src/results_store.md

```diff
@@ -0,0 +1,74 @@
+# Results
+
+Reefmod Engine Outputs and scenario information are recorded from the ReefMod Engine API
+after each scenario(s) run, not before.
+
+## ResultStore Struct
+
+The results stores holds all model outputs from both counterfactuals and interventions.
+Information about scenarios is stored in the scenario field of the result store. The reps
+field the result store is exactly have the number of scenarios contained in the scenario
+dataframe and results YAX dataset. This is because the api forces counterfactuals to be
+evaluated with every intervention run.
+
+Model inputs are stored in the`results` field of the store and contains the following
+variables:
+
+ - `total_cover`      ~ Total Coral Cover (% of total reef area)
+ - `dhw`              ~ Degree Heating Weeks (°C weeks)
+ - `dhw_mortality`    ~ DHW Mortality (% of population*)
+ - `cyc_mortality`    ~ Cyclone Mortality (% of population*)
+ - `cyc_cat`          ~ Cyclone Category
+ - `cots`             ~ COTS Population (per ha)
+ - `cots_mortality`   ~ Mortality caused by COTS (% of population*)
+ - `total_taxa_cover` ~ Total Species Cover (% of total reef area)
+
+*\* not formally confirmed*
+
+## Usage
+
+A side effect of the c++ API structure means that each intervention scenario must be
+executed separately (**Intervention scenario** meaning a run with a specific intervention
+strategy, not referring to runs with differing environmental inputs called repeats).
+
+1. Before storing any results, create the results struct.
+```julia
+result_store = ResultStore(start_year, end_year)
+```
+2. Perform model run
+
+```julia
+...
+
+# Initializing Run
+@RME runInit()::Cint
+
+# Starting Run
+@RME runProcess()::Cint
+```
+
+3. Store results
+
+`reps` is the number of repeats executed in the run.
+```julia
+concat_results!(result_store, start_year, end_year, reps)
+```
+
+## Saving Results
+
+Results can be saved using `save_result_store`.
+```julia
+save_result_store(<result_store>, <dir_name>)
+```
```

Comment:
```suggestion
Results can be saved using `save_result_store`.

```julia
save_result_store(<result_store>, <dir_name>)
```
```

Reply:
Oh I just realised:

The convention in Julia is to have paths first, followed by object.

For example:

```julia
df = DataFrame(x=1, y=2)
CSV.write(output_dir, df)

using FileIO
save("test.png", rand(Gray, 100, 100))
```

Suggest we align with that to avoid unexpected surprises compared to other Julia packages.

Reply:
Also `save_result_store()` seems to be unexported